skimage/feature/scattering.py

```diff
@@ -0,0 +1,286 @@
+    else:
+        J = len(wavelet_filters['psi'][0])  # number of scales
+        L = len(wavelet_filters['psi'][0][0])  # number of orientations
+    num_signals = x.shape[0]
+    # 3.- Check the Order (m) of the scattering transform, can only be 0,1,2,
+    # and that gives us different number of scattering coefficients
+    num_coefs = {
+        0: int(1),
+        1: int(1 + J * L),
+        2: int(1 + J * L + J * (J - 1) * L ** 2 / 2)
+    }.get(m, -1)
+    if num_coefs == -1:
+        warning_string = "Parameter m out of bounds, " \
+                         "valid values are 0,1,2 not {0}"
+        warnings.warn(warning_string.format(m))
+        return
+    # constants
+    spatial_coefs = int(x.shape[1]/2**(J-1))
+    # subsample at a rate a bit lower than the critic frequency
+    oversample = 1
+    U = []
+    V = []
+    v_resolution = []
+    current_res = 0
+    # vars to return (where we save the scattering and its accessing
+    # tree-structure (a dictionary)
+    S = np.ndarray((num_signals, num_coefs, spatial_coefs, spatial_coefs))
+    # allows access to the coefficients (S) using the tree structure
+    S_tree = {}
+    # Start computing the scattering coefficients
+    # Zero order scattering coeffs
+    S[:, 0, :, :] = _apply_lowpass(x, wavelet_filters['phi'][current_res],
+                                   J,  spatial_coefs)
+    S_tree[0] = S[:, 0, :, :]
+    l_indexing = np.arange(0, L)
+    # First order scattering coeffs
+    if m > 0:
+        Sview = S[:, 1:J*L+1, :, :].view()
+        Sview.shape = (num_signals, J, L, spatial_coefs, spatial_coefs)
+        # precompute the fourier transform of the images
+        X = np.fft.fft2(x)
+        for j in np.arange(J):
+            filtersj = wavelet_filters['psi'][current_res][j].view()
+            # resolution for the next layer
+            resolution = max(j-oversample, 0)
+            v_resolution.append(resolution)
+            # fft2(| x conv Psi_j |): X is full resolution,
+            # as well as the filters
+            aux = _subsample(np.fft.ifft2(_apply_fourier_mult(X, filtersj)),
+                             resolution)
+            V.append(aux)
+            U.append(np.abs(aux))
+            filter = wavelet_filters['phi'][resolution]
+            Sview[:, j, :, :, :] = _apply_lowpass(U[j],
+                                                  filter, J,
+                                                  spatial_coefs)
+            j_indexing = np.ones((L))*int(j)
+            for i_l, first_order_label in \
+                    enumerate(zip(j_indexing, l_indexing)):
+                S_tree[first_order_label] = Sview[:, j, i_l, :, :]
+    # Second order scattering coeffs
+    if m > 1:
+        sec_order_coefs = int(J*(J-1)*L**2/2)
+        S2norder = S[:, J*L+1:num_coefs, :, :]  # view of the data
+        S2norder.shape = (num_signals, int(sec_order_coefs/L),
+                          L, spatial_coefs, spatial_coefs)
+        indx = 0
```

Comment:
If this is an abbreviation for the word "index", please add the missing character :)